torch/_meta_registrations.py

```diff
@@ -1206,14 +1232,75 @@ def _check_foreach_binop_tensor_lists(self, other):
     )
 
 @register_meta([aten._foreach_add.List])
-def meta__foreach_add(self, other, alpha):
+def meta__foreach_add(self, other, alpha=1):
     _check_foreach_binop_tensor_lists(self, other)
     return [torch.empty_like(s) for s in self]
 
 
 @register_meta([aten._foreach_add_.List])
-def meta__foreach_add_(self, other, alpha):
+def meta__foreach_add__list(self, other, alpha=1):
+    _check_foreach_binop_tensor_lists(self, other)
+
+
+@register_meta([aten._foreach_div_.List])
+def meta__foreach_binop__list(self, other):
+    _check_foreach_binop_tensor_lists(self, other)
+
+
+@register_meta([aten._foreach_mul.List])
+def meta__foreach_binop_list(self, other):
     _check_foreach_binop_tensor_lists(self, other)
+    return [torch.empty_like(s) for s in self]
+
+
+@register_meta(
+    [
+        aten._foreach_add_.Scalar,
+        aten._foreach_mul_.Scalar,
+        aten._foreach_sub_.Scalar,
+    ]
+)
+def meta__foreach_binop__scalar(self, scalar):
+    check(
+        isinstance(self, List),
+        f"The first argument of must be List[Tensor], but got {type(self)}.",
+    )
+
+
+@register_meta([aten._foreach_div.Scalar])
+def meta__foreach_binop_scalar(self, scalar):
+    check(
+        isinstance(self, List),
+        f"The first argument of must be List[Tensor], but got {type(self)}.",
+    )
+    return [torch.empty_like(s) for s in self]
+
+
+@register_meta(
+    [
+        aten._foreach_addcdiv_.Scalar,
+        aten._foreach_addcmul_.Scalar,
+    ]
+)
+def meta__foreach_addcop__scalar(self, tensor1, tensor2, scalar=1):
+    assert all([isinstance(l, List) for l in [self, tensor1, tensor2]]), (
+        "All arguments of _foreach_addcmul_ must be List[Tensor], but got "
```

Comment:
Nit: don't hardcode `addcmul_` into the error string if you're re-using this meta kernel for multiple ops.

If we need the hardcoded string (to satisfy tests), you'll have to register the function separately.